chromadb/api/models/AsyncCollection.py

```diff
@@ -307,6 +311,29 @@ async def upsert(
             uris=uris,
         )
 
+    async def batch_upsert(
+        self,
+        ids: IDs,
+        embeddings: Optional[Union[Embeddings, List[np.ndarray]]] = None,  # type: ignore
+        metadatas: Optional[List[Optional[Metadata]]] = None,
+        documents: Optional[List[Optional[Document]]] = None,
+        images: Optional[List[Optional[Image]]] = None,
+        uris: Optional[List[Optional[URI]]] = None,
+        batch_size: int = 1024,
+        progress_callback: Optional[Callable[[IDs], None]] = None,
```

Comment:
open to other names for this parameter; maybe `on_batch_upserted`?

Reply:
that feels clearer in "when its called" and less clear on "how its intended to be used". I think naming for "when its called" is probably better as maybe there are other uses